### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by Uthmanbello**
## Highlights
- Wrote tests for `factorial` and `reverse` methods
- Created `Solver` class
- Defined `factorial` and `reverse` methods
- Wrote tests for `fizzbuzz` method
- Defined `fizzbuzz` method
- Added readme file

### CODE REVIEW

**Review by bobb-Rob — COMMENTED** [review]
Hi @Uthmanbello and @dfsalam , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has some failed linter checks. Please fix and re-submit.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'rspec'
+require './solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'returns factorial of a given number' do
+      expect(subject.factorial(5)).to eql(120)
+    end
+  end
```
Comment (binhussen) [inline spec/solver_spec.rb]: - Great Job for testing Positive value :+1: I think it would be better if you have tests for all cases in TDD, here for factorial you can add tests for `Negative value` and `0 value` which will return 0

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Uthmanbello , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- All linter checks passed ✔️
- Followed TDD ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (Uthmanbello) [reply spec/solver_spec.rb]: Hello @binhussen ,

Thanks a lot for the review. However, I think the factorial of `0 value` returns `1`

Thanks

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
